channeld/channeld.c

```diff
@@ -944,6 +944,25 @@ static struct bitcoin_signature *unraw_sigs(const tal_t *ctx,
 	return sigs;
 }
 
+/* Do we want to update fees? */
+static bool want_fee_update(const struct peer *peer, u32 *target)
+{
+	u32 max;
+
+	if (peer->channel->opener != LOCAL)
+		return false;
+
+	max = approx_max_feerate(peer->channel);
+	*target = peer->desired_feerate;
+
+	/* FIXME: We should avoid adding HTLCs until we can meet this
+	 * feerate! */
+	if (*target > max)
+		*target = max;
+
+	return *target != channel_feerate(peer->channel, REMOTE);
```

Comment:
I guess this behavior is probably defined in a BOLT somewhere but shouldn't we be using *our* desired feerate, negotiating with the peer to keep within *their* min and max, and also ensuring we never go above *our* max? Rather than just using *their* desired feerate (for a commitment tx *we're* paying for, for Pete's sake!) and merely ensuring that it doesn't exceed our max feerate?

Reply:
As funder, we set the feerates; the comparison here is simply "the feerate they are currently using".

We don't know the peer's allowable range; we just know what we want, and what we can afford (approx_max_feerate).

Reply:
Oh, I see. That makes sense :).